recipes/LJSpeech/S2ST/Quantization/train.py

```diff
@@ -0,0 +1,155 @@
+"""
+Script to train K-means clustering model
+
+Authors
+ * Jarod Duret 2023
+"""
+
+
+import sys
+import logging
+import time
+
+import joblib
+import torch
+from sklearn.cluster import MiniBatchKMeans
+from hyperpyyaml import load_hyperpyyaml
+import speechbrain as sb
+from speechbrain.lobes.models.huggingface_wav2vec import HuggingFaceWav2Vec2
+
+from utils import extract_features, get_splits
+
+
+def setup_logger():
+    log_format = "[%(asctime)s] [%(levelname)s]: %(message)s"
+    logging.basicConfig(format=log_format, level=logging.INFO)
+    logger = logging.getLogger(__name__)
+    return logger
+
+
+def get_device(use_cuda):
+    use_cuda = use_cuda and torch.cuda.is_available()
+    print("\n" + "=" * 30)
+    print("USE_CUDA SET TO: {}".format(use_cuda))
+    print("CUDA AVAILABLE?: {}".format(torch.cuda.is_available()))
+    print("=" * 30 + "\n")
+    return torch.device("cuda" if use_cuda else "cpu")
+
+
+def fetch_kmeans_model(
+    n_clusters,
+    init,
+    max_iter,
+    batch_size,
+    tol,
+    max_no_improvement,
+    n_init,
+    reassignment_ratio,
+    random_state,
+):
+    return MiniBatchKMeans(
+        n_clusters=n_clusters,
+        init=init,
+        max_iter=max_iter,
+        batch_size=batch_size,
+        tol=tol,
+        max_no_improvement=max_no_improvement,
+        n_init=n_init,
+        reassignment_ratio=reassignment_ratio,
+        random_state=random_state,
+        verbose=1,
+        compute_labels=True,
+        init_size=None,
+    )
+
+
+def train_kmeans(kmeans_model, features_batch):
+    start_time = time.time()
+    kmeans_model.fit(features_batch)
+    time_taken = round((time.time() - start_time) // 60, 2)
+    return kmeans_model, time_taken
+
+
+if __name__ == "__main__":
+    logger = setup_logger()
+
+    # Load hyperparameters file with command-line overrides
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # Create experiment directory
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
+    sys.path.append("../../")
+    from ljspeech_prepare import prepare_ljspeech
+
+    sb.utils.distributed.run_on_main(
+        prepare_ljspeech,
+        kwargs={
+            "data_folder": hparams["data_folder"],
+            "save_folder": hparams["save_folder"],
+            "splits": hparams["splits"],
+            "seed": hparams["seed"],
+            "skip_prep": hparams["skip_prep"],
+        },
+    )
+
+    # Fetch device
+    device = get_device(not hparams["no_cuda"])
+
+    logger.info(f"Loading encoder model from HF hub: {hparams['encoder_hub']}")
+    save_path = f"{hparams['save_folder']}/pretrained_models"
+    encoder = HuggingFaceWav2Vec2(
```

Comment:
why now using yaml here like for other recipes?

Reply:
Done